test/mount_test.py

```diff
@@ -71,15 +72,26 @@ def condition(fn):
     assert repr(Path(local_dir)) in repr(m)
 
 
-def test_create_mount_file_errors(servicer, tmpdir, client):
-    m = Mount._from_local_dir(Path(tmpdir) / "xyz", remote_path="/xyz")
-    with pytest.raises(FileNotFoundError):
+def test_create_mount_file_errors(servicer, tmp_path, client):
+    invalid_dir = tmp_path / "xyz"
+    m = Mount._from_local_dir(invalid_dir, remote_path="/xyz")
```

Comment:
Could we write the tests to assert about the public API (`Image.add_local_file` etc.) rather than the internal `Mount` object? Now that the latter is private it's possible we'll change the implementation for the public `Image` API and it's more important to be sure that we get the user-facing errors we expect from the public methods.